index.html

```diff
@@ -594,33 +586,38 @@ <h3><dfn>MonetizationStartEventInit</dfn> dictionary</h3>
               <dt><dfn>requestId</dfn> member</dt>
               <dd>
                 The Monetization ID (UUID V4) generated by the browser (see
-                <a href="#flow">Flow</a>).
+                [[[#flow]]]).
               </dd>
             </dl>
           </section>
         </section>
         <section
           data-dfn-for="MonetizationProgressEvent"
-          data-link-for="MonetizationProgressEvent"
         >
           <h2><dfn>MonetizationProgressEvent</dfn> interface</h2>
           <p>
             Dispatched every time an ILP packet with a non-zero amount has been
-            fulfilled by the page&apos;s Web Monetization receiver (including
-            the first time, when
-            <code>monetizationstart</code> is also dispatched). This event MAY
-            NOT be emitted if there are no listeners for it on
-            <code>document.monetization</code>.
+            fulfilled by the page's Web Monetization receiver (including
+            the first time, when `monetizationstart` event is also dispatched).
           </p>
           <pre class="idl">
-              [Constructor(DOMString type, optional MonetizationProgressEventInit eventInitDict = {}), SecureContext, Exposed=Window]
-              interface MonetizationProgressEvent : Event {
-                readonly attribute DOMString amount;
-                readonly attribute DOMString assetCode;
-                readonly attribute DOMString assetScale;
-              };
-            </pre
-          >
+            [SecureContext, Exposed=Window]
+            interface MonetizationProgressEvent : Event {
+              constructor(DOMString type, optional MonetizationProgressEventInit eventInitDict = {});
```

Comment:
Why `optional` ?

Reply:
None of the `MonetizationStartEventInit` members are required ^_^

Reply:
Yet amount/assetCode/assetScale attributes are "readonly" ?

Reply:
I don't have any experience with Web IDL so I could be confused, but how else besides a constructor could you set "readonly" attributes ?

I see also that the paymentPointer and requestId (referred sometimes to as session id iirc) are missing, not inline with the example from the "jump to the code" section ?

See: https://github.com/adrianhopebailie/web-monetization/blob/master/explainer.md#handle-payments

Reply:
> but how else besides a constructor could you set "readonly" attributes

That's a slightly different problem (but you are not wrong!) - but basically, `MonetizationStartEventInit` members would either need to be made required or have defaults: then eventInitDict can be non-optional. So, we need to decide which route to take... I think they need to be required.